lib/Client.js

```diff
@@ -114,6 +114,7 @@ class Client extends EventEmitter {
     * @arg {Number | String} [options.gateway.maxShards=1] The total number of shards you want to run. If "auto" Dysnomia will use Discord's recommended shard count.
     * @arg {Function} [options.gateway.reconnectDelay] A function which returns how long the bot should wait until reconnecting to Discord.
     * @arg {Boolean} [options.gateway.seedVoiceConnections=false] Whether to populate bot.voiceConnections with existing connections the bot account has during startup. Note that this will disconnect connections from other bot sessions
+    * @arg {String} [options.gateway.gatewayURL] A custom gateway url to allow the use of gateway proxies
```

Comment:
Although I'm not explicitly against this being added, I believe the description needs a bit of rewording. My concerns are following:

- This feels more like a last resort and/or a debugging option rather than something to be used in production – going from the assumption that most people who proxy the gateway also more than likely proxy REST, it should be overridden at the REST proxy level by overriding the autodiscovery endpoint response rather than forcing it here.
- This feature adds a possibility to implement behavior regarded as an anti-pattern (someone could force `wss://gateway.discord.gg/` as their gateway URL instead of using the autodiscovery mechanism)

```suggestion
    * @arg {String} [options.gateway.gatewayURL] Overrides the gateway URL that Dysnomia will use to connect to Discord's gateway
```